### ISSUE TITLE
CPP error due to incorrect relative path in `-I` ghc flag

### ISSUE THREAD

**Issue by jneira** [mpickering/hie-bios#168]
* In the lhs repo, with actual master and removing the `hie.yaml` inside the submodule `ghcide`
* `hie-bios check .\ghcide\src\Development\IDE\Core\Compile.hs` throw the error `hie-bios.exe: `gcc.exe' failed in phase `C pre-processor'. (Exit code: 1)`
  * orignally detected when loading the file using `haskell-language-server` lsp server
* `hie-bios debug .\ghcide\src\Development\IDE\Core\Compile.hs`: 
```
Root directory:      D:\dev\ws\haskell\hls
GHC options:         -fbuilding-cabal-package -O0 -outputdir D:\dev\ws\haskell\hls\dist-newstyle\build\x86_64-windows\ghc-8.6.5\ghcide-0.1.0\build -odir D:\dev\ws\haskell\hls\dist-newstyle\build\x86_64-windows\ghc-8.6.5\ghcide-0.1.0\build -hidir D:\dev\ws\haskell\hls\dist-newstyle\build\x86_64-windows\ghc-8.6.5\ghcide-0.1.0\build -stubdir D:\dev\ws\haskell\hls\dist-newstyle\build\x86_64-windows\ghc-8.6.5\ghcide-0.1.0\build -i -iD:\dev\ws\haskell\hls\dist-newstyle\build\x86_64-windows\ghc-8.6.5\ghcide-0.1.0\build -iD:\dev\ws\haskell\hls\ghcide\src -iD:\dev\ws\haskell\hls\dist-newstyle\build\x86_64-windows\ghc-8.6.5\ghcide-0.1.0\build\autogen -iD:\dev\ws\haskell\hls\dist-newstyle\build\x86_64-windows\ghc-8.6.5\ghcide-0.1.0\build\global-autogen -ID:\dev\ws\haskell\hls\dist-newstyle\build\x86_64-windows\ghc-8.6.5\ghcide-0.1.0\build\autogen -ID:\dev\ws\haskell\hls\dist-newstyle\build\x86_64-windows\ghc-8.6.5\ghcide-0.1.0\build\global-autogen -ID:\dev\ws\haskell\hls\dist-newstyle\build\x86_64-windows\ghc-8.6.5\ghcide-0.1.0\build -Iinclude -ID:\dev\ws\haskell\hls\dist-newstyle\build\x86_64-windows\ghc-8.6.5\ghcide-0.1.0\build\include -optP-include -optPD:\dev\ws\haskell\hls\dist-newstyle\build\x86_64-windows\ghc-8.6.5\ghcide-0.1.0\build\autogen\cabal_macros.h -this-unit-id ghcide-0.1.0-inplace -hide-all-packages -Wmissing-home-modules -no-user-package-db -package-db C:\sd\ghc-8.6.5\package.db -package-db D:\dev\ws\haskell\hls\dist-newstyle\packagedb\ghc-8.6.5 -package-db D:\dev\ws\haskell\hls\dist-newstyle\build\x86_64-windows\ghc-8.6.5\ghcide-0.1.0\package.conf.inplace -package-id Win32-2.6.1.0 -package-id aeson-1.4.7.0-cd3226f3592c14b57317868b05ca111c8406e823 -package-id array-0.5.3.0 -package-id async-2.2.2-8a92aac7185731110d2c49fa4c5e178c57db5776 -package-id base-4.12.0.0 -package-id binary-0.8.6.0 -package-id bytestring-0.10.8.2 -package-id containers-0.6.0.1 -package-id data-default-0.7.1.1-7013baa9116df0b50d31a3c8eb903b25839e440c -package-id deepseq-1.4.4.0 -package-id directory-1.3.3.0 -package-id extra-1.7.1-928e18ccfde73619dca99476fb8157a60567deba -package-id filepath-1.4.2.1 -package-id fuzzy-0.1.0.0-17d1fc92109ad41018dca4e6d0c3d5e889db6a70 -package-id ghc-8.6.5 -package-id ghc-boot-8.6.5 
-package-id ghc-boot-th-8.6.5 -package-id haddock-libra_-1.8.0-f92be63798189d509a672751c8d1214bca5ee1a7 -package-id hashable-1.3.0.0-86175f0b9a4f8f4556fe1f3cbacca6fee53ede99 -package-id haskell-lsp-0.21.0.0-b0dd6a6db6fa49d0c0d8a6f3e1f11f88d79edb52 
-package-id haskell-lsp-t_-0.21.0.0-cb99083a4328b1c2df15d01e5af54d29048ae8a8 -package-id hslogger-1.3.1.0-5b3cd30dad1b594162d526108ad13197bbe2ed86 -package-id mtl-2.2.2 -package-id network-uri-2.6.3.0-33e9ca334997e0b7c95df50e052acac566452dba -package-id prettyprinter-1.6.1-b137a4980eb7ae62b7fcc649493fb9e211272034 -package-id prettyprinter_-1.1.1.2-0c791b2ee4cae39cb06e2e2702f6451116f54ae3 -package-id regex-tdfa-1.3.1.0-17c6ed1d2f9cb588ce9dd04b173467e7b90fcd0e -package-id rope-utf16-sp_-0.3.1.0-c819382868cb68ee44d8273637377238f6474a82 -package-id safe-exceptio_-0.1.7.0-b1a63dc7fe7db051df518023e185a4d8c61bc088 -package-id shake-0.18.5-9d2964d163c9c836b91d0fa6b1d14ad46c1fdeb8 -package-id sorted-list-0.2.1.0-23e5efc8c0a10c4c450e600e73fe6e73bcd70108 -package-id stm-2.5.0.0 -package-id syb-0.7.1-48e3cd8b6ed49ac179bbcd19eccd3692465b6fc1 -package-id text-1.2.3.1 -package-id time-1.8.0.2 -package-id transformers-0.5.6.2 -package-id unordered-con_-0.2.10.0-a8c36b9adfe8096e31a7f35c3d03f461a61841c5 -package-id utf8-string-1.0.1.1-03035c961e9261cf14a303008766c17c1d25ae92 -XHaskell2010 -XBangPatterns -XDeriveFunctor -XDeriveGeneric -XGeneralizedNewtypeDeriving -XLambdaCase -XNamedFieldPuns -XOverloadedStrings -XRecordWildCards -XScopedTypeVariables -XStandaloneDeriving -XTupleSections -XTypeApplications -XViewPatterns Development.IDE.Compat Development.IDE.Core.Debouncer Development.IDE.Core.FileStore Development.IDE.Core.IdeConfiguration Development.IDE.Core.OfInterest Development.IDE.Core.PositionMapping Development.IDE.Core.Rules Development.IDE.Core.RuleTypes Development.IDE.Core.Service Development.IDE.Core.Shake Development.IDE.GHC.Error Development.IDE.GHC.Util Development.IDE.Import.DependencyInformation Development.IDE.LSP.HoverDefinition Development.IDE.LSP.LanguageServer Development.IDE.LSP.Outline Development.IDE.LSP.Protocol Development.IDE.LSP.Server Development.IDE.Spans.Common Development.IDE.Types.Diagnostics Development.IDE.Types.Location Development.IDE.Types.Logger Development.IDE.Types.Options Development.IDE.Plugin Development.IDE.Plugin.Completions Development.IDE.Plugin.CodeAction Development.IDE.Core.Compile Development.IDE.Core.Preprocessor Development.IDE.Core.FileExists Development.IDE.GHC.Compat Development.IDE.GHC.CPP Development.IDE.GHC.Orphans Development.IDE.GHC.Warnings Development.IDE.GHC.WithDynFlags Development.IDE.Import.FindImports Development.IDE.LSP.Notifications Development.IDE.Spans.AtPoint Development.IDE.Spans.Calculate Development.IDE.Spans.Documentation Development.IDE.Spans.Type Development.IDE.Plugin.CodeAction.PositionIndexed Development.IDE.Plugin.CodeAction.Rules Development.IDE.Plugin.CodeAction.RuleTypes Development.IDE.Plugin.Completions.Logic Development.IDE.Plugin.Completions.Types -Wall -Wno-name-shadowing -hide-all-packages
System libraries:    D:\bin\stack\x86_64-windows\ghc-8.6.5\lib
Config Location:     D:\dev\ws\haskell\hls\hie.yaml
Cradle:              Cradle {cradleRootDir = "D:\\dev\\ws\\haskell\\hls", cradleOptsProg = CradleAction: Cabal}
Dependencies:        haskell-language-server.cabal cabal.project
```
* if we call ghc with the compiler options:
```
D:\\dev\\ws\\haskell\\hls\\ghcide\\src\\Development\\IDE\\Core\\Rules.hs:10:2: fatal error: ghc-api-version.h: No such file or directory

  ^
compilation terminated.
`gcc.exe' failed in phase `C pre-processor'. (Exit code: 1)
```
* we can observe that compiler flags contains `-Iinclude` when the missing header file is in `.\ghcide\include` so it works if we changed the flag to `-Ighcide\include`

**Comment by fendor** [comment]
Should be fixed by #166

**Comment by fendor** [comment]
Should be fixed, please reopen if it isnt :)
